frontend/src/metabase/reference/databases/databases.integ.spec.js

```diff
@@ -0,0 +1,156 @@
+import {
+    login,
+    createTestStore
+} from "metabase/__support__/integrated_tests";
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { CardApi } from 'metabase/services'
+
+import { 
+    FETCH_DATABASE_METADATA,
+    FETCH_DATABASES
+} from "metabase/redux/metadata";
+
+import { END_LOADING } from "metabase/reference/reference"
+
+import DatabaseListContainer from "metabase/reference/databases/DatabaseListContainer";
+import DatabaseDetailContainer from "metabase/reference/databases/DatabaseDetailContainer";
+import TableListContainer from "metabase/reference/databases/TableListContainer";
+import TableDetailContainer from "metabase/reference/databases/TableDetailContainer";
+import TableQuestionsContainer from "metabase/reference/databases/TableQuestionsContainer";
+import FieldListContainer from "metabase/reference/databases/FieldListContainer";
+import FieldDetailContainer from "metabase/reference/databases/FieldDetailContainer";
+
+import DatabaseList from "metabase/reference/databases/DatabaseList";
+import List from "metabase/components/List.jsx";
+import ListItem from "metabase/components/ListItem.jsx";
+import ReferenceHeader from "../components/ReferenceHeader.jsx";
+import AdminAwareEmptyState from "metabase/components/AdminAwareEmptyState.jsx";
+
+describe("The Reference Section", () => {
+    // Test data
+    const cardDef = { name :"A card", display: "scalar", 
+                      dataset_query: {database: 1, table_id: 1, type: "query", query: {source_table: 1, "aggregation": ["count"]}},
+                      visualization_settings: {}}
+
+    // Scaffolding
+    beforeAll(async () => {
+        await login();
+
+    })
+
+    describe("The Data Reference for the Sample Database", async () => {
+        
+        // database list
+        it("should see a single database", async ()=>{
+            const store = await createTestStore()
+            store.pushPath("/reference/databases/");
+            var container = mount(store.connectContainer(<DatabaseListContainer />));
+            await store.waitForActions([FETCH_DATABASES, END_LOADING])
+            
+            expect(container.find(ReferenceHeader).length).toBe(1)
+            expect(container.find(DatabaseList).length).toBe(1)            
+            expect(container.find(AdminAwareEmptyState).length).toBe(0)
+            
+            expect(container.find(List).length).toBe(1)
+            expect(container.find(ListItem).length).toBe(1)
+        })
+        
+        // database detail
+        it("should see a the detail view for the sample database", async ()=>{
```

Comment:
this should probably check that dom contains some expected information about the database?